TidepoolKit/Internal/WeakSynchronizedSet.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  WeakSynchronizedSet.swift
+//  TidepoolKit
+//
+//  Created by Michael Pangburn
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+
+/// A set-like collection of weak types, providing closure-based iteration on a client-specified queue
+/// Mutations and iterations are thread-safe
+public class WeakSynchronizedSet<Element> {
```

Comment:
Hm...maybe not make this `public`?  We don't want name collisions...

Reply:
Ah, good point.